datafusion/common/src/stats.rs

```diff
@@ -70,3 +72,24 @@ pub struct ColumnStatistics {
     /// Number of distinct values
     pub distinct_count: Option<usize>,
 }
+
+impl ColumnStatistics {
+    pub fn is_singleton(&self) -> bool {
```

Comment:
I recomment a docstring explaning what a singleton is (e.g. that the column contains a single, non null value)